packages/shortest/src/browser/core/browser-tool.ts

```diff
@@ -875,4 +875,84 @@ export class BrowserTool extends BaseBrowserTool {
       },
     );
   }
+
+  /**
+   * Waits for DOM to stabilize
+   * DOM is considered stabilized when:
+   * - DOMContentLoaded is fired
+   * - No mutations are detected for 1 second (e.g new elements such as modals, popups, etc.)
+   */
+  public async waitForStableDOM(): Promise<void> {
+    const page = this.page;
+    if (!page) {
+      throw new Error("No page found.");
+    }
+    try {
+      await this.waitForDOMContentLoaded();
+      await page.evaluate(async () => {
+        return await new Promise<void>((resolve) => {
+          const createTimeout = () => {
+            return setTimeout(() => {
+              resolve();
+              observer.disconnect();
+            }, 1000);
```

Comment:
It looks like we use `1000` in 2 places (here and line 925). Can be extracted to a constant?